exo-go/src/template_helpers/index.go

```diff
@@ -167,3 +170,49 @@ func CreateTmpServiceDir(chosenTemplate string) string {
 	}
 	return serviceTmpDir
 }
+
+// AddTemplate fetches a remote template from GitHub and stores it
+// under templateDir, returns an error if any
+func AddTemplate(gitURL, templateName, templateDir string) error {
+	if output, err := processHelpers.Run(fmt.Sprintf("git submodule add --name %s %s %s", templateName, gitURL, templateDir), ""); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to fetch template from %s: %s\n", gitURL, output))
+	}
+	return nil
+}
+
+// FetchTemplates fetches updates for all existing remote templates
+func FetchTemplates() error {
+	if output, err := processHelpers.Run("git submodule foreach git pull origin master", ""); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to fetch updates for existing templates: %s\n", output))
+	}
+	return nil
+}
+
+// RemoveTemplate removes the given template from the application
+func RemoveTemplate(templateName, templateDir string) error {
+	if output, err := processHelpers.Run(fmt.Sprintf("git submodule deinit -f %s", templateDir), ""); err != nil {
```

Comment:
The way we call commands right now is cumbersome:
- first we concatenate words into strings using `fmt.Sprintf`
- then in processHelpers we parse the string back into words

Let's provide the command to run directly as as an array of words here. When using varargs, this is intuitive in Go:

```go
... = processHelpers.Run("git", "submodule", "deinit", "-f", templateDir) ...
```

The definition of the Run method:

```go
func Run(command ...string)   // command is an array of strings